pkg/autodiscovery/providers/docker.go

```diff
@@ -52,11 +52,13 @@ func (d *DockerConfigProvider) String() string {
 // Collect retrieves all running containers and extract AD templates from their labels.
 func (d *DockerConfigProvider) Collect() ([]integration.Config, error) {
 	var err error
+	firstCollection := false
 	if d.dockerUtil == nil {
```

Comment:
Note: This code is not thread safe as 2 threads can run line 56 at the same time and then run line 57.